### PR TITLE
Add DP-GEN Structure.md to doc

### PR DESCRIPTION
**Pull request by HuangJiameng**
A basic user guide.

### CODE REVIEW

**Review by njzjz — COMMENTED** [review]
The file needs to be added to TOC in `index.rst`, otherwise it will not be shown in the website.

**Comment by HuangJiameng** [comment]
> The file needs to be added to TOC in `index.rst`, otherwise it will not be shown in the website.

OK, it is added.

File: doc/overview/DP-GEN-structure.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+├── arginfo.py
+├── ch4
+├── __init__.py
+├── lib
+└── run.py
+````
+run.py is the core of DP-GEN, corresponding to `dpgen run`, we can find `make_train`, `run_train`, ... `post_fp`, and other steps related functions here.
+
+## Find how a parameter is used in the code
+It is strongly recommended that you use the `find in files` function of Virtual Studio software, or similar functions of other software. Type in the name of the parameter you're looking for, and you'll see where it's read in, and the procedure used.
```
Comment (njzjz) [inline doc/overview/DP-GEN-structure.md:55]: Do you mean **Visual** Studio?

File: doc/overview/DP-GEN-structure.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+Of course, you can also search for the relevant code according to the above guide.
+
+## Want to modify a function?
+If you have special requirements, you can make personalized modifications in the code corresponding to the function. If you think your modification can benefit the public, and it does not conflict with the current DP-GEN function; or if you fix a bug, please make a pull requist to contribute the optimization to the DP-GEN repository.
```
Comment (njzjz) [inline doc/overview/DP-GEN-structure.md:59]: requist -> request

**Review by njzjz — CHANGES_REQUESTED** [review]

**Review by njzjz — APPROVED** [review]

**Pull request merged**